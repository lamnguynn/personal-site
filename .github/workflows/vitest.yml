name: Run unit and UI tests on project

on:
    push:
      branches:
        - "main"
    pull_request:

jobs:
  run-vitest:
    name: Run vitest
    runs-on: ubuntu-latest
    steps:
      - name:  Checkout code
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
      - name: Setup Node
        uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4.4.0
        with:
          node-version: 24
      - name: Install dependencies
        run: |
            npm ci
      - name: Run unit tests
        run: |
          npm run test-unit
      - name: Run UI tests
        run: |
          npm run test-ui