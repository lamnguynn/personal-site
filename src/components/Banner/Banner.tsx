import type { ReactNode } from 'react';

interface Props {
  message: ReactNode;
}

export default function Banner({ message }: Props) {
  return (
    <div className="absolute z-[100000000] flex h-fit w-full items-center justify-center bg-red-400 p-2">
      {message}
    </div>
  );
}
