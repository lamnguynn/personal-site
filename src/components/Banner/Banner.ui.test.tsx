import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import Banner from './Banner';

describe('Banner component', () => {
  test('snapshot', () => {
    const { asFragment } = render(
      <Banner message={<p>Sample banner message</p>} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
