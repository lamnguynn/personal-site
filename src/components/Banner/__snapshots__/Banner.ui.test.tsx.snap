// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Banner component > snapshot 1`] = `
<DocumentFragment>
  <div
    class="absolute z-[100000000] flex h-fit w-full items-center justify-center bg-red-400 p-2"
  >
    <p>
      Sample banner message
    </p>
  </div>
</DocumentFragment>
`;
